Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BPMNEditorController.java

```diff
@@ -231,6 +232,39 @@ public void onEvent(final Event event) throws InterruptedException {
       }
     });
 
+    this.addEventListener("onSimulateModel", new EventListener<Event>() {
+      @Override
+      public void onEvent(final Event event) throws InterruptedException {
+        PortalContext portalContext = mainC.getPortalContext();
+        Map<String, PortalPlugin> portalPluginMap = portalContext.getPortalPluginMap();
+        PortalPlugin simulateModelPlugin = portalPluginMap.get(PluginCatalog.PLUGIN_SIMULATE_MODEL);
+
+        //Since simulate model is an EE feature, it may not be available
+        if (simulateModelPlugin == null) {
+          Messagebox.show(Labels.getLabel("portal_simModelUnavailable_message"),
+                  Labels.getLabel("portal_simModelUnavailable_title"),
+                  Messagebox.OK, Messagebox.INFORMATION);
+          return;
+        }
+
+        if (currentUserAccessType == AccessType.VIEWER) {
+          Notification.error(Labels.getLabel("portal_noPrivilegeSaveEdit_message"));
+          return;
+        }
+
+        //Only allow model to be simulated if it is saved
+        if (isNewProcess || process == null) {
+          Notification.error(Labels.getLabel("portal_saveModelFirst_message"));
+        } else {
+          Map arg = new HashMap<>();
+          arg.put("selectedModel", process);
+          arg.put("modelData", eventToString(event));
+          simulateModelPlugin.setSimpleParams(arg);
+          simulateModelPlugin.execute(portalContext);
```

Comment:
The setSimpleParams looks like not thread-safe and it won't work properly in a multi-user environment. For example, a user calls this plugin while another user calls this plugin concurrently.